src/Microsoft.Azure.SignalR.Management/Negotiation/DefaultEndpointGetter.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    internal class DefaultEndpointGetter : IEndpointGetter
+    {
+        private readonly IServiceEndpointManager _serviceEndpointManager;
+        private readonly IEndpointRouter _router;
+        private readonly string _hubName;
+
+        public DefaultEndpointGetter(IServiceEndpointManager serviceEndpointManager, IEndpointRouter router, string hubName)
```

Comment:
where to specify endpoints for each function?

Reply:
Negotiation is a little different from sending messages. We just generates connection info for each endpoints.